pkg/ipamd/ipamd.go

```diff
@@ -538,37 +550,20 @@ func (c *IPAMContext) nodeInit() error {
 		vpcV4CIDRs = c.updateCIDRsRulesOnChange(vpcV4CIDRs)
 	}, 30*time.Second)
 
-	eniConfigName, err := eniconfig.GetNodeSpecificENIConfigName(ctx, c.cachedK8SClient)
-	if err == nil && c.useCustomNetworking && eniConfigName != "default" {
-		// Signal to VPC Resource Controller that the node is using custom networking
-		err := c.SetNodeLabel(ctx, vpcENIConfigLabel, eniConfigName)
-		if err != nil {
-			log.Errorf("Failed to set eniConfig node label", err)
-			podENIErrInc("nodeInit")
-			return err
-		}
-	} else {
-		// Remove the custom networking label
-		err := c.SetNodeLabel(ctx, vpcENIConfigLabel, "")
-		if err != nil {
-			log.Errorf("Failed to delete eniConfig node label", err)
-			podENIErrInc("nodeInit")
-			return err
+	if c.enablePodENI {
+		if c.useCustomNetworking {
+			if _, err = eniconfig.GetNodeSpecificENIConfigNameAndSendEvent(ctx, c.cachedK8SClient, c.eventRecorder, true); err != nil {
+				return err
+			}
 		}
-	}
 
-	if metadataResult.TrunkENI != "" {
-		// Signal to VPC Resource Controller that the node has a trunk already
-		err := c.SetNodeLabel(ctx, "vpc.amazonaws.com/has-trunk-attached", "true")
-		if err != nil {
-			log.Errorf("Failed to set node label", err)
-			podENIErrInc("nodeInit")
-			// If this fails, we probably can't talk to the API server. Let the pod restart
-			return err
+		if metadataResult.TrunkENI != "" {
+			// Signal event to VPC RC that the node has a trunk already
```

Comment:
Don't we want to signal the event to VPC RC about `trunk attached = true` or since it is found in metadataResult assumption is the node is already managed in VPC RC? If it is not needed let's remove the comment.

Reply:
Right, I think we don't have to send a confirmation back to VPC CNI since CNI has it' own check on trunk interface. I will remove the meaningless comment here after all comments are collect for the current revision. Thanks @jayanthvn

Reply:
But, this is done only once on bootup isn't it? Also, how will VPC RC rebuild the cache upon a restart?

Reply:
Do you mean VPC CNI check trunk? In current code, VPC CNI does reconcile this every 30s in update IP resources and rely on datastore for trunk info. Upon restart of VPC RC, it iterates through all nodes and manages all nodes which have valid labels. Their trunk interfaces are also described and cached.

Reply:
Right now, yes. But once we transition to events based approach, VPC RC would've to go through the events against the node to do the same, isn't it?

Reply:
Once we change to event, RC relies on event and node label. The workflow looks like this

1. CNI emits events continuously till finding trunk interface on node 
2. RC reconcile events and label node if event is valid
3. RC provision interfaces based on node label
4. if RC restarts, RC looks for node label for building cache

If events are still alive during RC booting up, RC will 

1. if the node is in cache, RC will skip the node
2. if the node hasn't been added in cache, RC will retry to label the node which should be safe